content/ui-patterns/saving.mdx

```diff
@@ -0,0 +1,266 @@
+
+## Explicit saving
+
+When designing a form, start with an explicit saving pattern. Avoid mixing explicit and automatic save patterns on a single page with multiple forms, and never mix save patterns in a single form. For example: if I upload a profile photo in the "Public profile" form, that change should not be saved until I explicitly submit the whole form.
+
+Explicitly saved forms can be saved by clicking a submit button or by pressing Enter while focused on an input.
+
+If there is an error saving the data, the user's data should be preserved in the form and they should be given [feedback](#feedback) about the failure.
+
+<DoDontContainer>
+  <Do>
+    <img alt="A form broken into two sections where one section has a dropdown that automatically saves" src="https://user-images.githubusercontent.com/2313998/151887935-980adacd-41aa-47cb-8756-eeaa1a6d0f2b.png" />
+    <Caption>Separate auto-saving inputs from an explicitly saved form</Caption>
+  </Do>
+  <Dont>
+    <img alt="A form with a single section and it contains a dropdown that automatically saves" src="https://user-images.githubusercontent.com/2313998/151887936-d41900ca-2d63-43ac-bf82-55b4bba0df61.png" />
+    <Caption>Don't mix auto-saving inputs and explicitly saved inputs in the same form</Caption>
+  </Dont>
+</DoDontContainer>
+
+### Unsaved changes
+
+If a user tries to navigate away from a page with a form that has unsaved changes, you have the option to warn them that their changes will be lost using the [`beforeunload` event](https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event). Modern browsers prevent developers from customizing the `alert` message, so we're limited to a generic message like "Do you want to leave this site? Changes you made may not be saved."
+
+### Declarative controls
+
+To ensure your forms are accessible, you should use explicit saving (and not automatic saving) in declarative inputs. These are:
+
+- Text inputs
+- Checkbox groups
+- Radio button groups
+- Browser-native `<select>` dropdown menus
+- Multi-select dropdown menus
+
+To ensure your forms are accessible, do not automatically save declarative inputs.
+
+There are a few issues to bear in mind with autosaving declarative inputs.
+
+- Text inputs: Users expect to set a value and then submit. Sensitive information could be unintentionally submitted, such as a change password input.
+- Checkbox groups: It could be unclear whether or not the selections has been saved. Users may accidentally submit a selection clicked by accident.
+- Radio button groups: Screenreader users can't read the options without selecting them, which could accidentally apply a selection the user didn't want.
+- Browser-native `<select>` dropdowns: Similarly to radio button groups, options could be selected when a user focuses the select and uses the arrow keys to read through each option.
```

Comment:
I would expect this guidance to be on the `Automatic saving` section, to explain scenarios where that choice might not be advisable.